### ISSUE TITLE
Easy improvement: Add a start label

### ISSUE THREAD

**Issue by EngandDeveloper** [CSchank/finsm#96]
When creating finite state machines it would be good to being able to add start label for the start state. Right now (as far as I know) user has to go to the LaTeX code and modify the first state and add initial keyword. I believe adding a start label wouldn't be too hard since in current version we can click the "F" to label it as final, I believe it is possible to do it with "S" key to add start label.

Ps: this is what I mean by start label (marked with red).
![image](https://user-images.githubusercontent.com/46913754/75701588-09509000-5c82-11ea-84ed-136cc40a78e7.png)

**Comment by CSchank** [comment]
The TAs know that bold means start but yeah, maybe we'll talk about this for consistency.

**Comment by CSchank** [comment]
@Necried what do you think? I'm thinking we might want to 180 on this and change in favour of consistency.

**Comment by Necried** [comment]
Good point on consistency. Things to think about:
1. Positioning of the "Start" label. Automated? Let users position it?
2. Does this get replicated correctly when exported to Latex?
3. Does this get complicated when it comes to multiple start labels for NFAs?

If we have satisfactory plans for each of the above, I won't have a problem with this.

**Comment by CSchank** [comment]
@Necried, here are some thoughts I have on your questions:
1. LaTeX by default positions it on the left, so we could just force them to put it there. The downside of this is that they would have to maybe change the design of their machine to accommodate. I'm sure we could allow them to rotate it around the state kind of like self-loops (but without being able to change the length/distance), but I'd have to look into what the TikZ api supports.
2. Yes, I don't think we'll have issues, as long as ours looks fairly similar in the app.
3. @EngandDeveloper Remind me, does Dr. Farmer show multiple start labels for NFAs with more than one start state? Perhaps you haven't quite gotten there.

Anyways @EngandDeveloper a bit of background: we actually purposely didn't include these for many of the reasons @Necried and I are discussing here. We also wanted it to be consistent with the way we do the simulation (with bold). The TAs and Dr. Farmer are aware of this, but I agree it's a confusing point, and if some people are going in and editing the LaTeX code, then we should seriously consider changing this so they don't feel the need to do so. 

@Necried the other option is we don't render it like this until they export the LaTeX, but I don't really like that idea. If we do this, we should add a shortcut (perhaps "S") to set states as start states in Build mode too.

**Comment by CSchank** [comment]
Okay, my research (see http://www.texample.net/media/pgf/builds/pgfmanualCVS2012-11-04.pdf, pg. 315/316) tells me that you can position the arrows either `above`, `below`, `left`, or `right` of the states:
![image](https://user-images.githubusercontent.com/12722104/75713370-aff35b80-5c97-11ea-8724-28f3302b64eb.png)

@Necried we could allow these 4 directions?

**Comment by CSchank** [comment]
This will be slightly tricky, since we're actually going to need more information in our machine model, and we'll have to be very careful not to break people's currently-stored machines. But it will be good for us to practice how to migrate data properly. We'll have to test a lot before deploying. @EngandDeveloper very little in life is "simple" unfortunately 😉Thoughts from either of you?

**Comment by EngandDeveloper** [comment]
We haven't seen any example which has multiple start points. However, might see them as the course continues. Thank you so much for putting the time and effort to search about it. I think it is not as simple as I thought :) I started the issue since it might be a problem when you design the transition diagram on the website and then when you add the start label sometimes it messes up your design by overlapping with edges.

**Comment by CSchank** [comment]
Indeed. What do you think? Is rotating the arrows a much-needed feature? Or could we always default them to the left?

**Comment by Necried** [comment]
Defaulting to the left would make things consistent. The burden will fall on our users to accommodate their state machines to ensure that it is "displayed" properly.

I do like the idea of rotation, but as you've already said it will be more complicated than it seems. It could be worth implementing @CSchank.

@EngandDeveloper since you're an active user of the app currently, ignoring implementation difficulties, which design choice appeals to you?

**Comment by EngandDeveloper** [comment]
I think rotating might come handy in some situations, but as long as the arrow's length can be changed default left start arrow would be fine.
